octree/include/pcl/octree/impl/octree_search.hpp

```diff
@@ -50,11 +48,16 @@ bool
 OctreePointCloudSearch<PointT, LeafContainerT, BranchContainerT>::voxelSearch(
     const PointT& point, Indices& point_idx_data)
 {
-  assert(isFinite(point) &&
```

Comment:
I think the `isFinite` checks should also stay in `assert`. Usually the caller (another PCL algorithm, e.g. a segmentation method) would already know that the point has valid coordinates before calling `voxelSearch`/`nearestKSearch`/...